### PR TITLE
Add unit tests for layout

### LINKED ISSUE #20: Add unit tests to FilteredBooleanQuery
It currently lacks tests, add them.

### PR DESCRIPTION
**Pull request by trescube**
Fixed #20 

Unit tests did not exist and it helped in my understanding of how the code works so I wrote them.

### CODE REVIEW

File: layout/FilteredBooleanQuery.js [hunk 1]
```diff
@@ -53,9 +53,6 @@ Layout.prototype.render = function( vs ){
   // handle sorting views under 'sort'
   if( this._sort.length ){
     this._sort.forEach( function( view ){
-      if( !Array.isArray( q.sort ) ){
```
Comment (dianashk) [inline layout/FilteredBooleanQuery.js:56]: was this no longer necessary? i'm not sure what it was doing to begin with, but interesting to understand why it isn't needed.

File: package.json [hunk 2]
```diff
@@ -18,9 +19,11 @@
   },
   "homepage": "https://github.com/pelias/query#readme",
   "dependencies": {
-    "check-types": "^7.0.0"
+    "check-types": "^7.0.0",
+    "jshint": "^2.9.2"
```
Comment (dianashk) [inline package.json]: should this be in both dependencies and devDependencies and with different versions?

File: layout/FilteredBooleanQuery.js (hunk 1 above)
Comment (trescube) [inline layout/FilteredBooleanQuery.js:56]: it's already an array, the check is redundant

File: package.json (hunk 2 above)
Comment (trescube) [inline package.json]: This is actually a carry over from https://github.com/pelias/query/pull/19, not sure how it ended up in both places.

File: view/phrase.js [hunk 3]
```diff
@@ -5,14 +5,14 @@ module.exports = function( vs ){
   if( !vs.isset('input:name') ||
       !vs.isset('phrase:analyzer') ||
       !vs.isset('phrase:field') ||
-      !vs.isset('phrase:boost'),
+      !vs.isset('phrase:boost') ||
```
Comment (dianashk) [inline view/phrase.js:8]: 🏆

**Comment by dianashk** [comment]
Look at all those beautiful tests!!!!

File: package.json [hunk 4]
```diff
@@ -18,9 +19,11 @@
...
   },
   "devDependencies": {
+    "jshint": "^2.5.6",
```
Comment (orangejulius) [inline package.json:26]: if you run `npm install -D precommit-hook && npm install`, precommit-hook will automatically generate precommit hooks that run the unit tests and lint before every git commit, to make sure they are passing. We use it in most of our other repos so we should do it here too.

Comment (trescube) [inline package.json:26]: K, I can do that.

File: view/popularity.js [hunk 5]
```diff
@@ -9,7 +9,7 @@ module.exports = function( subview ){
 
     // validate required params
     if( !vs.isset('popularity:field') ||
-        !vs.isset('popularity:modifier'),
+        !vs.isset('popularity:modifier') ||
```
Comment (orangejulius) [inline view/popularity.js:12]: oh wow this happened all over!

**Comment by orangejulius** [comment]
So many tests and linting :+1:

**Pull request merged**